packages/smarthr-ui/src/components/Dropdown/SortDropdown/useSortDropdown.ts

```diff
@@ -75,10 +81,24 @@ export const useSortDropdown = ({ sortFields, defaultOrder, onApply, decorators
     (e) => {
       const select = e.currentTarget
       const newLabel = select.options[select.selectedIndex].label
-      const newFields = innerFields.map((field) => ({
-        ...field,
-        selected: field.label === newLabel,
-      }))
+      const newFields = innerFields.map((field) => {
+        if (field.label === newLabel) {
+          if (!field.selected) {
+            return {
+              ...field,
+              selected: true,
+            }
+          }
+        } else if (field.selected) {
+          return {
+            ...field,
+            selected: false,
+          }
+        }
+
+        return field
+      })
```

Comment:
必ず配列内すべてのオブジェクトを置き換えていましたが、必要な場合のみ再生成するようにロジックを変更しています。

Reply:
ifとelse ifと早期returnでやりたいことがわかりづらかったので、以下のような感じだとどうでしょう？

```suggestion
      const newFields = innerFields.map((field) => {
        if (field.label === newLabel && !field.selected) {
          return field
        }
        
        return {
          ...field,
          selected: !field.selected
        }
      })
```

Reply:
↑だと条件ミスってますね...

Reply:
↑のコードはこれはこれで条件全部読まないとよくわからないんですよねぇ...
パターンは3つだけだしMapperにしますか

Reply:
個人的にはこれくらいのifのネストは許容範囲かな〜と思いますが↑も良さそうなんで取り込んじゃいますね

Reply:
ありがとうございます！
見てても分かる通りちょっと自分はelse ifとreturnが組み合わさった条件分岐をきちんと読み解ける自信がないので笑、助かります！！